# HG changeset patch
# User Rik <rik@octave.org>
# Date 1389736040 28800
#      Tue Jan 14 13:47:20 2014 -0800
# Node ID 87381dbbe25ee039e0d10db860cf5b86519758fe
# Parent  a0c9f5302eaef020d7cdecfc815612c488990b59
Clear all var, functions, globals when clearing classes (bug #41229)

* variables.cc (do_matlab_compatible_clear, Fclear): Add
symbol_table::clear_all () to case branch that clears classes.

diff --git a/libinterp/corefcn/variables.cc b/libinterp/corefcn/variables.cc
--- a/libinterp/corefcn/variables.cc
+++ b/libinterp/corefcn/variables.cc
@@ -2252,16 +2252,17 @@ do_matlab_compatible_clear (const string
         {
           symbol_table::clear_variables ();
         }
       else if (argv[idx] == "classes"
                && ! symbol_table::is_local_variable ("classes"))
         {
           symbol_table::clear_objects ();
           octave_class::clear_exemplar_map ();
+          symbol_table::clear_all ();
         }
       else
         {
           symbol_table::clear_symbol_pattern (argv[idx]);
         }
     }
 }
 
@@ -2461,16 +2462,17 @@ without the dash as well.\n\
                   else if (clear_variables)
                     {
                       do_clear_variables (argv, argc, idx, exclusive);
                     }
                   else if (clear_objects)
                     {
                       symbol_table::clear_objects ();
                       octave_class::clear_exemplar_map ();
+                      symbol_table::clear_all ();
                     }
                   else
                     {
                       do_clear_symbols (argv, argc, idx, exclusive);
                     }
                 }
 
               octave_link::set_workspace ();
